packages/next/src/lib/metadata/metadata.tsx

```diff
@@ -55,24 +56,42 @@ export function createMetadataComponents({
   async function MetadataTree() {
     const defaultMetadata = createDefaultMetadata()
     let metadata: ResolvedMetadata | undefined = defaultMetadata
-    try {
-      const resolvedMetadata = await resolveMetadata({
-        tree,
-        parentParams: {},
-        metadataItems: [],
-        searchParams,
-        getDynamicParamFromSegment,
-        errorConvention: errorType === 'redirect' ? undefined : errorType,
-      })
+    let error: any
+    const errorMetadataItem: [null, null] = [null, null]
+    const errorConvention = errorType === 'redirect' ? undefined : errorType
 
-      // Skip for redirect case as for the temporary redirect case we don't need the metadata on client
-      if (errorType === 'redirect') {
-        metadata = defaultMetadata
-      } else {
-        metadata = await accumulateMetadata(resolvedMetadata, metadataContext)
-      }
+    const metadataResult = await resolveMetadata({
```

Comment:
wdyt about naming these properties, so it's a bit easier to follow what's happening in subsequent lines? eg:

```ts
const [resolvedMetadata, resolvedError] = await resolveMetadata({ ... })
```